p2p/node_info.go

```diff
@@ -24,22 +25,16 @@ type NodeInfo struct {
 
 // CompatibleWith checks if two NodeInfo are compatible with eachother.
 // CONTRACT: two nodes are compatible if the major version matches and network match
-// and they have at least one channel in common.
 func (info *NodeInfo) CompatibleWith(other *NodeInfo) error {
-	iMajor, _, _, err := splitVersion(info.Version)
-	if err != nil {
-		return err
-	}
-	oMajor, _, _, err := splitVersion(other.Version)
+	compatible, err := version.CompatibleWith(other.Version)
```

Comment:
compatible, err := version.CompatibleWith(other.Version)
	if err != nil {
		return err
	}

	if !compatible {
		return fmt.Errorf("Peer is on a different major version. Peer version: %v, node version: %v.", other.Version, info.Version)
	}